scanpy/_utils.py

```diff
@@ -546,6 +546,14 @@ def check_presence_download(filename: Path, backup_url):
         from .readwrite import _download
         _download(backup_url, filename)
 
+def check_presence_download_untar(filename: Path, tarfilename: Path, backup_url):
+    """Check if file is present otherwise download and untar."""
+    if not filename.is_file():
+        if not tarfilename.is_file():
+            from .readwrite import _download
+            _download(backup_url, tarfilename)
+        import tarfile
+        tarfile.open(tarfilename).extractall(tarfilename.parents[0])
```

Comment:
```suggestion
        tarfile.open(tarfilename).extract(filename, tarfilename.parents[0])
```

Does this make sense? This way not everything is extracted. Please check if it works as intended.

Reply:
Nice idea. But the code line does not work because tar file is a folder called 'spatial' and intended files (and other files) are located in that folder. I change in a way that only targeted files are extracted (within /spatial). The following only extract targeted files within any structure of folders:
```suggestion
        tar = tarfile.open(tarfilename)
        _file = [_ for _ in tar.getnames() if Path(_).name == filename.name][0]
        tar.extract(_file , tarfilename.parent)```

Reply:
Nice idea. But the code line does not work because tar file is a folder called 'spatial' and intended files (and other files) are located in that folder. I change in a way that only targeted files are extracted (within /spatial). The following only extract targeted files within any structure of folders:
```suggestion
        tar = tarfile.open(tarfilename)
        _file = [_ for _ in tar.getnames() if Path(_).name == filename.name][0]
        tar.extract(_file , tarfilename.parent)```